tests/loss/test_dino_loss.py

```diff
@@ -0,0 +1,271 @@
+        return total_loss
+
+    @typing.no_type_check
+    @torch.no_grad()
+    def update_center(self, teacher_output):
+        """
+        Update center used for teacher output.
+        """
+        batch_center = torch.sum(teacher_output, dim=0, keepdim=True)
+        batch_center = batch_center / len(teacher_output)
+        # ema update
+        self.center = self.center * self.center_momentum + batch_center * (
+            1 - self.center_momentum
+        )
+
+
+class TestDINOLoss:
+    @pytest.mark.parametrize("batch_size", range(1, 4))
+    @pytest.mark.parametrize("n_local", range(0, 4))
+    @pytest.mark.parametrize("output_dim", range(1, 4))
+    def test_different_input_sizes(
+        self, batch_size: int, n_local: int, output_dim: int
+    ) -> None:
+        _run_test(batch_size=batch_size, n_local=n_local, output_dim=output_dim)
+
+    @pytest.mark.parametrize("warmup_teacher_temp", [0.01, 0.04, 0.07])
+    @pytest.mark.parametrize("teacher_temp", [0.01, 0.04, 0.07])
+    @pytest.mark.parametrize("warmup_teacher_temp_epochs", [0, 1, 10])
+    @pytest.mark.parametrize("epoch", [0, 1, 10, 20])
+    def test_teacher_temperature_warmup(
+        self,
+        warmup_teacher_temp: float,
+        teacher_temp: float,
+        warmup_teacher_temp_epochs: int,
+        epoch: int,
+    ) -> None:
+        _run_test(
+            warmup_teacher_temp=warmup_teacher_temp,
+            teacher_temp=teacher_temp,
+            warmup_teacher_temp_epochs=warmup_teacher_temp_epochs,
+            epoch=epoch,
+        )
+
+    @pytest.mark.parametrize("student_temp", [0.05, 0.1, 0.2])
+    @pytest.mark.parametrize("center_momentum", [0.5, 0.9, 0.95])
+    def test_other_parameters(
+        self,
+        student_temp: float,
+        center_momentum: float,
+    ) -> None:
+        _run_test(
+            student_temp=student_temp,
+            center_momentum=center_momentum,
+        )
+
+
+def test_center__equivalence() -> None:
+    """Check that DINOLoss.update_center is equivalent to Center.update.
+
+    TODO(Guarin, 08/24): Remove this test once DINOLoss uses Center internally.
+    """
+    criterion = DINOLoss(output_dim=32, center_momentum=0.9)
+    center = Center(size=(1, 1, 32), momentum=0.9)
+    x = torch.rand(2, 32)
+    criterion.update_center(teacher_out=x)
+    center.update(x=x)
+    assert torch.allclose(criterion.center, center.value)
+
+
+def _generate_output(
+    batch_size: int = 2, n_views: int = 3, output_dim: int = 4, seed: int = 0
+) -> List[Tensor]:
+    """Returns a list of view representations.
+
+    Example output:
+        [
+            torch.Tensor([img0_view0, img1_view0]),
+            torch.Tensor([img0_view1, img1_view1])
+        ]
+
+    """
+    torch.manual_seed(seed)
+    out = []
+    for _ in range(n_views):
+        views = [torch.rand(output_dim) for _ in range(batch_size)]
+        out.append(torch.stack(views))
+    return out
+
+
+def _run_test(
```

Comment:
I would explain better what this function does, at least with the name.
```suggestion
def _assert_dino_loss_equal_to_original(
```

Reply:
I think you were the original author and the function was named `test` but ok let me rename it.

Reply:
It was named `def _assert_dino_loss_equal_to_original(` in the original test, you changed it when copy-pasting it.

Reply:
Feel free to resolve.